src/query/sql/src/planner/optimizer/cascades/cascade.rs

```diff
@@ -29,36 +30,36 @@ use crate::optimizer::cost::DefaultCostModel;
 use crate::optimizer::format::display_memo;
 use crate::optimizer::memo::Memo;
 use crate::optimizer::rule::TransformResult;
-use crate::optimizer::RuleSet;
 use crate::optimizer::SExpr;
 use crate::IndexType;
-use crate::MetadataRef;
 
 /// A cascades-style search engine to enumerate possible alternations of a relational expression and
 /// find the optimal one.
 pub struct CascadesOptimizer {
-    pub(crate) memo: Memo,
-    pub(crate) cost_model: Box<dyn CostModel>,
+    pub memo: Memo,
+    pub explore_rule_set: roaring::RoaringBitmap,
+
+    pub cost_model: Box<dyn CostModel>,
+
     /// group index -> best cost context
-    pub(crate) best_cost_map: HashMap<IndexType, CostContext>,
-    pub(crate) explore_rule_set: RuleSet,
-    pub(crate) metadata: MetadataRef,
+    pub best_cost_map: HashMap<IndexType, CostContext>,
+    _ctx: Arc<dyn TableContext>,
```

Comment:
Introducing `TableContext` in `CascadesOptimizer` may not be a good idea. Additionally, I noticed that this context has been named as `_ctx`. If we do not intend to use it, perhaps it would be better to remove it?